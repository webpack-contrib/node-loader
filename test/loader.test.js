/**
 * @jest-environment node
 */
import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getModuleSource,
  getWarnings,
  readAsset,
} from "./helpers";

describe("loader", () => {
  it("should work", async () => {
    const compiler = getCompiler("simple.js", {
      name: "[name].[ext]",
      flags: 1,
    });
    const stats = await compile(compiler);

    expect(
      getModuleSource("./example/build/Release/hello.node", stats)
    ).toMatchSnapshot("module");
    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('should throw an error on broken "node" addon', async () => {
    const compiler = getCompiler("broken.js", {
      name: "[name].[ext]",
    });
    const stats = await compile(compiler);

    expect(getModuleSource("./broken.node", stats)).toMatchSnapshot("module");

    let error;

    try {
      execute(readAsset("main.bundle.js", compiler, stats));
    } catch (nodeLoaderError) {
      error = nodeLoaderError.message
        .replace(new RegExp(process.cwd(), "g"), "/absolute/path/to")
        .replace(/\\/g, "/");
    }

    expect(error).toBeDefined();
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
