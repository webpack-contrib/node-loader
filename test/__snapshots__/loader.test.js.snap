// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should throw an error on broken "node" addon: errors 1`] = `Array []`;

exports[`loader should throw an error on broken "node" addon: module 1`] = `
"
try {
  process.dlopen(module, __dirname + require(\\"path\\").sep + __webpack_public_path__ + \\"broken.node\\");
} catch (error) {
  throw new Error('node-loader:\\\\n' + error);
}
"
`;

exports[`loader should throw an error on broken "node" addon: warnings 1`] = `Array []`;

exports[`loader should work: errors 1`] = `Array []`;

exports[`loader should work: module 1`] = `
"
try {
  process.dlopen(module, __dirname + require(\\"path\\").sep + __webpack_public_path__ + \\"hello.node\\", 1);
} catch (error) {
  throw new Error('node-loader:\\\\n' + error);
}
"
`;

exports[`loader should work: result 1`] = `
Object {
  "default": "world",
}
`;

exports[`loader should work: warnings 1`] = `Array []`;
